sample_metadata/parser/generic_metadata_parser.py

```diff
@@ -49,6 +50,49 @@ def __init__(
         self.qc_meta_map = qc_meta_map or {}
         self.reads_column = reads_column
         self.gvcf_column = gvcf_column
+        self.allow_extra_files_in_search_path = allow_extra_files_in_search_path
+
+    async def validate_rows(
+        self, sample_map: Dict[str, Union[dict, List[dict]]]
+    ) -> bool:
+        if not self.reads_column:
+            return True
+
+        if not self.allow_extra_files_in_search_path:
+            return self.check_files_covered_by_file_map(sample_map)
+
+        return True
+
+    def check_files_covered_by_file_map(self, sample_map: Union[dict, List[dict]]) -> bool:
+        """
+        Check that the files in the search_paths are completely covered by the sample_map
+        """
+        filenames = []
+        for sm in (sample_map if isinstance(sample_map, list) else [sample_map]):
+            for rows in sm.values():
+                if isinstance(rows, list):
+                    for r in rows:
+                        filenames.extend(r.get(self.reads_column, '').split(','))
+                else:
+                    filenames.extend(rows.get(self.reads_column, '').split(','))
```

Comment:
A minor one, but it might be a bit more clear if it implemented the way as for `sample_map`:
```suggestion
                for r in (rows if if isinstance(rows, list) else [rows]):
                    filenames.extend(r.get(self.reads_column, '').split(','))
```